File: kotlin-remix-run-router/karakum/inheritanceModifiers/modifyPropertyInheritance.js
import ts from "typescript"

export default (node) => {
    if (
        ts.isPropertyDeclaration(node)
        && ts.isIdentifier(node.name)
        && (
            (
                (
                    node.name.text === "status"
                    || node.name.text === "statusText"
                    || node.name.text === "data"
                )
                && node.parent
                && node.parent.name?.text === "ErrorResponseImpl"
            )
        )
    ) {
        return "override"
    }

    return null
}

File: kotlin-tauri-apps-api/karakum/src/webMain/kotlin/tauri/apps/api/karakum/inheritanceModifiers/modifyPropertyInheritance.kt
package tauri.apps.api.karakum.inheritanceModifiers

import arrow.core.raise.nullable
import io.github.sgrishchenko.karakum.extension.InheritanceModifierContext
import io.github.sgrishchenko.karakum.util.getParentOrNull
import typescript.Node
import typescript.isClassDeclaration
import typescript.isIdentifier
import typescript.isPropertyDeclaration

fun modifyPropertyInheritance(node: Node, context: InheritanceModifierContext) = nullable {
    nullable {
        ensure(isPropertyDeclaration(node))

        val name = node.name
        ensure(isIdentifier(name))

        nullable { // open
            ensure(
                name.text == "label"
                        || name.text == "listeners"
            )

            val classNode = ensureNotNull(node.getParentOrNull())
            ensure(isClassDeclaration(classNode))
            ensure(
                classNode.name?.text == "Webview"
                        || classNode.name?.text == "Window"
            )

            "open"
        }
    }
}
